plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -819,9 +819,10 @@ private List<List<String>> getPartitionValues(String databaseName, String tableN
 
         // Do parallel partition fetch.
         CompletionService<List<List<String>>> completionService = new ExecutorCompletionService<>(partitionsReadExecutor);
+        List<Future<?>> futures = new ArrayList<>(partitionSegments);
```

Comment:
This change is quite welcome.
Given that the change is pretty low-level and hard to reproduce, I'm wondering whether a specific test which would proxy the glue client could be used to ensure that the failure functionality is behaving as expected.

See `TestHiveGlueMetastore` (probably a separate test class would be better) and https://github.com/trinodb/trino/blob/71c88a9e1ac27f591c7eac18391e0bb513817701/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueTableOperationsInsertFailure.java#L69-L81 for scaffolding.

Reply:
I'm not sure that testing this behavior is important enough to justify adding a new test class that mocks the AWS client and instruments the background threads just to assert that cancellation occurs, given that the code itself definitely makes that guarantee clear (especially with the latest revision). If you still feel strongly about adding one though I can take the time to add one, I'm just less convinced of the value in doing so.

Reply:
@pettyjamesm indeed, I'm not sure either whether this behaviour should be specifically tested.

I tested it myself a simplified version of your change  and it is working on my end.